### ISSUE TITLE
Why does `skipWhile()` only work once?

### ISSUE THREAD

**Issue by timmolendijk** [baconjs/bacon.js#540]
Consider the following code snippet:

```javascript
var off = Bacon.fromBinder(function (sink) {
  $('#on').on('click', function () {
    sink(false);
  });
  $('#off').on('click', function () {
    sink(true);
  });
  return $.noop;
}).toProperty(true);

off.log();

Bacon.interval(500, 'update').
  skipWhile(off).
  log();
```

It won't log "update" until you click the `#on` button, after which it will start logging, as expected.

But why won't it switch back to skipping updates (and no logging) as soon as `#off` is clicked?

**Comment by phadej** [comment]
That's the idea `skipWhile`. You are probably looking for [`filter`](http://baconjs.github.io/api.html#observable-filter-property)

**Comment by timmolendijk** [comment]
Ah okay. I must say I had not expected that from reading the docs, but I see it now.

Too bad that `.filter(property)` seems to behave like `.map(property)` in that it does not wait for `property` to hold a value. Same issue as addressed in #539. I see how it is technically correct, but it makes it practically unusable because – in my use cases at least – there tends to be no guarantee that `property` has been "loaded" at the time of filtering.

**Comment by timmolendijk** [comment]
I may be able to work around this problem by means of an initial value, but that doesn't change my point about questioning the usefulness of this behavior.
